### ISSUE TITLE
Group and permission management suggestion for PnP provisioning engine

### ISSUE THREAD

**Issue by VesaJuvonen** [OfficeDev/PnP#726]
Cross reference from Yammer at - https://www.yammer.com/itpronetwork/#/Threads/show?threadId=524173527

Suggestions for engine improvements:

1. Be able to provision groups. 
 I saw that we can add users to the existing admins, members,owners and visitors but apparently I cant add new groups (with a permission level)
 2. Be able to provision permissions for pages/listitems/libraries based on the Security Element

**Comment by PaoloPia** [comment]
About this topic ... I already had a plan to add to the PnP Core Library some item-level security-oriented extension methods like: break permission inheritance, set item/document level permissions, etc.
We could then use those methods, together with those that are already in the Core Library, to complete this implementation. Is this in target for May release or for the June one (more reasonable and doable :) ...).

**Comment by levalencia** [comment]
Paolo, you are the XML guru but I was thinking something like this:

1. For adding New Groups to the sharepoint site.
Most of the times people use Permission Levels, but sometimes people uses granular permissions, so it should support both. If one is specified the second one is ignored.
http://screencast.com/t/p3zEewfo

2. For lists and libraries, I should be able to define the same, also permission level or granular
http://screencast.com/t/buO2CCwa5DpK

The scenario I am having is that I am creating a site to make reservations for company holiday rooms.  There is a list of Requests, where only Requestors can Add Items, but they shouldnt be able to delete or view other people's items.  And there is another list of reservations, where admins can contribute, but requestors can only view info.

The way we did it through CSOM was:
1. On the request list we broke inheritance, clear all permissions, then add the admins with contribute permission level, and the requestors only with Add Item (Granular Permission)
2. On the reservations list, we broke inheritance, clear all permissions, then add the admins with contribute permission levels, and requestors with read permission level.


and Finally 
3.
Same permission should be able to be set on specific items, in my scenario, I created an App Part to Approve/Reject a request, that page should be visible only to Admins, so the same applies here.


Thanks, keep up the good work
